internal/monitors/kubernetes/utils/podcache.go

```diff
@@ -0,0 +1,101 @@
+package utils
+
+import (
+	"reflect"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type podsSet map[types.UID]bool
+
+// PodCache is used for holding values we care about from a pod
+// for quicker lookup than querying the API for them each time.
+type PodCache struct {
+	namespacePodUIDCache map[string]podsSet
+	podUIDNamespaceCache map[types.UID]string
+	podUIDLabelCache     map[types.UID]labels.Set
+	podUIDORCache        map[types.UID][]metav1.OwnerReference
+}
+
+// NewPodCache creates a new minimal pod cache
+func NewPodCache() *PodCache {
+	return &PodCache{
+		namespacePodUIDCache: make(map[string]podsSet),
+		podUIDNamespaceCache: make(map[types.UID]string),
+		podUIDLabelCache:     make(map[types.UID]labels.Set),
+		podUIDORCache:        make(map[types.UID][]metav1.OwnerReference),
+	}
+}
+
+// IsCached checks if a pod was already in the cache, or if
+// the mapped values have changed. Returns true if no change
+func (pc *PodCache) IsCached(pod *v1.Pod) bool {
+	labelSet := labels.Set(pod.Labels)
+	cachedLabelSet := pc.podUIDLabelCache[pod.UID]
+	cachedNamespace := pc.podUIDNamespaceCache[pod.UID]
+	cachedOR := pc.podUIDORCache[pod.UID]
+
+	return reflect.DeepEqual(cachedLabelSet, labelSet) &&
+		(cachedNamespace == pod.Namespace) &&
+		(reflect.DeepEqual(cachedOR, pod.OwnerReferences))
+}
+
+// AddPod adds or updates a pod in cache
+func (pc *PodCache) AddPod(pod *v1.Pod) {
+	if _, exists := pc.namespacePodUIDCache[pod.Namespace]; !exists {
+		pc.namespacePodUIDCache[pod.Namespace] = make(map[types.UID]bool)
+	}
+	pc.namespacePodUIDCache[pod.Namespace][pod.UID] = true
+	pc.podUIDNamespaceCache[pod.UID] = pod.Namespace
```

Comment:
Turn this inside out so that you have a single map that has a key of the uid and values of a struct that contain all of the necessary pod info that we need.

Reply:
Make that struct the thing that you pass to the dimPropsForPod function instead of a combination of pod uid and this podcache.